pyvista/errors.py

```diff
@@ -1,19 +1,67 @@
-"""PyVista specific errors."""
-# flake8: noqa: F401
-
-# Import accessible under `pyvista.errors`
-from pyvista.core.errors import (
-    AmbiguousDataError,
-    DeprecationError,
-    MissingDataError,
-    NotAllTrianglesError,
-    PointSetCellOperationError,
-    PointSetDimensionReductionError,
-    PointSetNotSupported,
-    PyVistaDeprecationWarning,
-    PyVistaEfficiencyWarning,
-    PyVistaFutureWarning,
-    PyVistaPipelineError,
-    VTKVersionError,
-)
-from pyvista.plotting.errors import InvalidCameraError, RenderWindowUnavailable
+"""Deprecated utilities subpackage."""
+import importlib
+import inspect
+import warnings
+
+# Places to look for the utility
+_MODULES = [
+    'pyvista.core.errors',
+    'pyvista.plotting.errors',
+]
+
+
+def _try_import(module, name):
+    """Attempt to import a module."""
+    _module = importlib.import_module(module)
+    try:
+        feature = inspect.getattr_static(_module, name)
+        import_path = f'from {module} import {name}'
+    except AttributeError:
+        return None, None
+    return feature, import_path
+
+
+def __getattr__(name):
+    """Fetch an attribute ``name`` from ``globals()`` and warn if it's from a deprecated module.
+
+    Note that ``__getattr__()`` only gets called when ``name`` is missing
+    from the module's globals. The trick is that we want to import this
+    function into other deprecated modules, and we want to carry this
+    subpackage's globals along to prevent some spurious warnings.
```

Comment:
Does this "import elsewhere" caveat still apply here? If not, the logic can be simplified I think, e.g. `globals()[name]` should never succeed as per this paragraph.